### PR TITLE
Fix precedencegroup and operator decl lookup

### PR DESCRIPTION
**Pull request by hamishknight**

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
Let's see what breaks

@swift-ci please test source compatibility

**Comment by hamishknight** [comment]
@swift-ci please test source compatibility

**Comment by hamishknight** [comment]
@swift-ci please test source compatibility

**Comment by hamishknight** [comment]
@swift-ci please test source compatibility

**Comment by hamishknight** [comment]
@swift-ci please test source compatibility

**Comment by hamishknight** [comment]
https://github.com/apple/llvm-project/pull/892

@swift-ci please test

**Comment by hamishknight** [comment]
https://github.com/apple/llvm-project/pull/892

@swift-ci please test

**Comment by hamishknight** [comment]
https://github.com/apple/llvm-project/pull/892

@swift-ci please test

**Comment by hamishknight** [comment]
https://github.com/apple/llvm-project/pull/892

@swift-ci please test

**Comment by hamishknight** [comment]
https://github.com/apple/llvm-project/pull/892

@swift-ci please test

**Comment by hamishknight** [comment]
https://github.com/apple/llvm-project/pull/892

@swift-ci please test

**Comment by hamishknight** [comment]
https://github.com/apple/llvm-project/pull/892

@swift-ci please test

**Comment by hamishknight** [comment]
https://github.com/apple/llvm-project/pull/892

@swift-ci please test

File: include/swift/Basic/NullablePtr.h [hunk 1]
```diff
@@ -37,7 +37,7 @@ class NullablePtr {
   template<typename OtherT>
   NullablePtr(NullablePtr<OtherT> Other,
               typename std::enable_if<
-                std::is_convertible<OtherT, T>::value,
+                std::is_convertible<OtherT *, T *>::value,
```
Comment (CodaFi) [inline include/swift/Basic/NullablePtr.h:40]: Let's split this one out separately.

**Pull request closed without merging**